### DISCUSSION TITLE
Custom Script Timeout on excel import

### DISCUSSION THREAD

**Discussion by Brescou** [netbox-community/netbox#15230] [Help Wanted!] [answered]
Hi!
I'm working on a custom script in NetBox to import data from an Excel file with over 16k lines, and I'm hitting the job timeout limit. Is there a way to handle large datasets without changing the global job_timeout setting? Any strategies for optimizing script performance or managing long-running tasks would be greatly appreciated!

**Comment by candlerb** [comment, marked as answer]
You can set the job_timeout on a per-script basis (under `class Meta`), without touching the global default.

https://docs.netbox.dev/en/stable/customization/custom-scripts/#job_timeout
